github/EnterpriseConsumedLicenses.py

```diff
@@ -0,0 +1,85 @@
+#                                                                              #
+# PyGithub is distributed in the hope that it will be useful, but WITHOUT ANY  #
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS    #
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more #
+# details.                                                                     #
+#                                                                              #
+# You should have received a copy of the GNU Lesser General Public License     #
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+from typing import Any, Dict, Union
+
+from github.GithubObject import Attribute, CompletableGithubObject, NotSet, _NotSetType
+from github.NamedEnterpriseUser import NamedEnterpriseUser
+from github.PaginatedList import PaginatedList
+
+
+class EnterpriseConsumedLicenses(CompletableGithubObject):
+    """
+    This class represents Enterprises. The reference can be found here https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/license#list-enterprise-consumed-licenses
+    """
+
+    def _initAttributes(self) -> None:
+        self._total_seats_consumed: Attribute[int] = NotSet
+        self._total_seats_purchased: Attribute[int] = NotSet
+        self._login: Attribute[str] = NotSet
+        self._url: Attribute[str] = NotSet
+
+    def __repr__(self) -> str:
+        return self.get__repr__({"login": self._login.value})
+
+    @property
+    def total_seats_consumed(self) -> int:
+        return self._total_seats_consumed.value
+
+    @property
+    def total_seats_purchased(self) -> int:
+        return self._total_seats_purchased.value
+
+    @property
+    def login(self) -> str:
+        self._completeIfNotSet(self._login)
+        return self._login.value
+
+    @property
+    def url(self) -> str:
+        self._completeIfNotSet(self._url)
+        return self._url.value
+
+    def get_enterprise_users(
+        self, filter_: Union[str, _NotSetType] = NotSet, role: Union[str, _NotSetType] = NotSet
+    ) -> PaginatedList[NamedEnterpriseUser]:
+        """
+        :calls: `GET /enterprise/{enterprise}/consumed-licenses <https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/license#list-enterprise-consumed-licenses>`_
+        """
+        assert filter_ is NotSet or isinstance(filter_, str), filter_
+        assert role is NotSet or isinstance(role, str), role
+
+        url_parameters = {}
+        if filter_ is not NotSet:
+            url_parameters["filter"] = filter_
+        if role is not NotSet:
+            url_parameters["role"] = role
+        return PaginatedList(NamedEnterpriseUser, self._requester, self.url, url_parameters, None, "users")
```

Comment:
That is weird, we are requesting the same url as in `Enterprise.get_consumed_licenses`.

From the docs, it looks like we retrieve the `EnterpriseConsumedLicenses` object with an `users` attributes, for which we have retrieved the first page. There is no other endpoint that behaves like that.

I think this requires some extension of PaginatedList, or we fetch the first page again when accessing the `users` attribute (as you propose).

Reply:
Thank you for the feedback! I also thought that's weird to request the same url for both of them. However, unfortunatelly, there are two types of responses in this API. When a user want to get just "total seat puchased", the user don't have to do pagenation because the value of total seat purchased is already in the first page of response. There is a wasted time while the pagenation roop. 
> or we fetch the first page again when accessing the users attribute (as you propose). 

That's why I proposed it. This kind of mixed result is rare case, I think we shouldn't update PaginatedList for this.

Reply:
Before we start extending PaginatedList, can I ask you to create some simple tests for `MainClass.get_enterprise` and `Enterprise.get_consumed_licenses`? For now without iterating over `users`? This requires recording test data, which will help me understand how exactly the pagination of `/enterprises/{enterprise}/consumed-licenses` might work. Please feel free to obfuscate enterprise specific content.

Reply:
I can create some simple tests but I actually don't have any enterprise ids for testing. Is there any of them for testing here? Or just create tests with blank on enterprise id?

Reply:
I thought you'd have access to some enterprise instance, otherwise, how would you know these changes work?

Reply:
I mean, I have our company own enterprise instance and I can access and test with it, but our company cannot give the access to not a company members.

Reply:
There is no need  to give access to any one, simply record the data for the tests against your enterprise instance, obfuscate data where really needed and add the test files to your pull request. Once recorded, nobody needs access to the targeted enterprise instance.

Reply:
Understand. I haven't been doing this kind of testing without actual api call, but will try to find out how to do this on.

Reply:
@EnricoMi Sorry for late. I created the data and script for testing about Enterprise class. Please check it with it. Thank you so much.